plugins/inputs/ceph/ceph.go

```diff
@@ -411,26 +432,53 @@ func decodeStatus(acc telegraf.Accumulator, input string) error {
 	return nil
 }
 
+// decodeStatusFsmap decodes the FS map portion of the output of 'ceph -s'
+func decodeStatusFsmap(acc telegraf.Accumulator, data *CephStatus) error {
+	fields := map[string]interface{}{
+		"up":         data.FSMap.NumUp,
+		"in":         data.FSMap.NumIn,
+		"max":        data.FSMap.NumMax,
+		"up_standby": data.FSMap.NumUpStandby,
+	}
+	acc.AddFields("ceph_fsmap", fields, map[string]string{})
+	return nil
+}
+
 // decodeStatusHealth decodes the health portion of the output of 'ceph status'
 func decodeStatusHealth(acc telegraf.Accumulator, data *CephStatus) error {
+	statusCodes := map[string]float64{
+		"HEALTH_ERR":  0,
+		"HEALTH_WARN": 1,
+		"HEALTH_OK":   2,
+	}
 	fields := map[string]interface{}{
 		"status":         data.Health.Status,
-		"overall_status": data.Health.OverallStatus,
+		"status_code":    statusCodes[data.Health.Status],
+		"overall_status": data.Health.OverallStatus, // This field is no longer reported in ceph 10 and later
 	}
 	acc.AddFields("ceph_health", fields, map[string]string{})
 	return nil
 }
 
+// decodeStatusMonmap decodes the Mon map portion of the output of 'ceph -s'
+func decodeStatusMonmap(acc telegraf.Accumulator, data *CephStatus) error {
+	fields := map[string]interface{}{
+		"num_mons": data.MonMap.NumMons,
+	}
+	acc.AddFields("ceph_monmap", fields, map[string]string{})
+	return nil
+}
+
 // decodeStatusOsdmap decodes the OSD map portion of the output of 'ceph -s'
 func decodeStatusOsdmap(acc telegraf.Accumulator, data *CephStatus) error {
 	fields := map[string]interface{}{
-		"epoch":            data.OSDMap.OSDMap.Epoch,
-		"num_osds":         data.OSDMap.OSDMap.NumOSDs,
-		"num_up_osds":      data.OSDMap.OSDMap.NumUpOSDs,
-		"num_in_osds":      data.OSDMap.OSDMap.NumInOSDs,
-		"full":             data.OSDMap.OSDMap.Full,
-		"nearfull":         data.OSDMap.OSDMap.NearFull,
-		"num_remapped_pgs": data.OSDMap.OSDMap.NumRemappedPGs,
+		"epoch":            data.OSDMap.Epoch,
+		"num_osds":         data.OSDMap.NumOSDs,
+		"num_up_osds":      data.OSDMap.NumUpOSDs,
+		"num_in_osds":      data.OSDMap.NumInOSDs,
+		"num_remapped_pgs": data.OSDMap.NumRemappedPGs,
+		"full":             data.OSDMap.Full,
+		"nearfull":         data.OSDMap.NearFull,
```

Comment:
I would also keep ordering here, so it becomes clear only the double OSDMap has been removed.
```suggestion
		"full":             data.OSDMap.Full,
		"nearfull":         data.OSDMap.NearFull,
		"num_remapped_pgs": data.OSDMap.NumRemappedPGs,
```

Reply:
Latest changes have all fields sorted alphabetically.

Previously, they matched the ordering of what was already in the plugin, or the output of the `ceph` commands, but this is probably easier overall going forward.